internal/coreinternal/processor/filterlog/filterlog.go

```diff
@@ -90,11 +90,11 @@ func NewMatcher(mp *filterconfig.MatchProperties) (Matcher, error) {
 // supported to have more than one of these specified, and all specified must
 // evaluate to true for a match to occur.
 func (mp *propertiesMatcher) MatchLogRecord(lr plog.LogRecord, resource pcommon.Resource, library pcommon.InstrumentationScope) bool {
-	if lr.Body().Type() == pcommon.ValueTypeString && mp.bodyFilters != nil && mp.bodyFilters.Matches(lr.Body().StringVal()) {
-		return true
+	if lr.Body().Type() == pcommon.ValueTypeString && mp.bodyFilters != nil && !mp.bodyFilters.Matches(lr.Body().StringVal()) {
+		return false
 	}
-	if mp.severityTextFilters != nil && mp.severityTextFilters.Matches(lr.SeverityText()) {
-		return true
```

Comment:
Why this didn't work for https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/11439 before?

Reply:
if the severity didn't match, it would simply go to the next check (attributes) - and this would match (empty expected attributes means it always matches).

I've added a missing test case that reproduced this behavior.

Reply:
Makes sense. Thanks for clarifying!